src/server.ts

```diff
@@ -27,14 +24,24 @@ app.post("/", function (req: any, res: any) {
 });
 
 // Put @ /{id}/done - marks a todo as done and returns a HTTP status code to indicate success/failure.
+// parseInt required as reqest are mostly string
 app.put("/:id/done", function (req: any, res: any) {
-	console.log(typeof req.params.id);
-	res.send(markTodoAsDone(parseInt(req.params.id)));
+	let result = markTodoAsDone(parseInt(req.params.id)); //positional so userid wont work
+	if (result === true) {
+		res.status(200).end(); //chaining//json,send,end goes to client not status
+	} else {
+		res.status(400).end();
+	}
 });
 
 // Delete @ /{id} - delete a todo item. Returns the deleted object and the appropriate HTTP status code.
 app.delete("/:id", function (req: any, res: any) {
-	res.send(removeTodo(req.params.id));
+	try {
+		res.json(removeTodoById(parseInt(req.params.id)));
+		res.status(200).end(); //chaining//json,send,end goes to client not status
```

Comment:
You shouldn't have both these lines. `res.json(removeTodoById(parseInt(req.params.id)));` will send the data with status code `200` to the user by default, so you don't need `res.status(200).end();`